core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/resourcePolicies/NaiveResourceAllocator.scala

```diff
@@ -0,0 +1,192 @@
+import edu.uci.ics.amber.engine.common.workflow.PhysicalLink
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos.WorkflowRuntimeStatistics
+import edu.uci.ics.texera.workflow.common.workflow.{PartitionInfo, PhysicalPlan}
+
+import scala.collection.mutable
+
+/**
+  * Allocates resources for a given region and its operators.
+  *
+  * @param region The region for which to allocate resources.
+  * @return A tuple containing:
+  *         1) A new Region instance with new configuration.
+  *         2) An estimated cost of the workflow with the new configuration,
+  *         represented as a Double value (currently set to 0, but will be
+  *         updated in the future).
+  * Notes:
+  * 1. If the count of in running historical records for one operator are
+  *    less than 3, there is no need to allocate the workers, because it is
+  *    too fast. ( < 3, the first in running record might has warmup. So we
+  *    need at least one useful in running record to get the speed/throughput)
+  * 2. The historical data needs to have the same version with the current
+  *    workflow.
+  * 3. The speed unit can be not exactly 1 second. It is more likely to represent
+  *    a ratio.
+  */
+class NaiveResourceAllocator(physicalPlan: PhysicalPlan, executionClusterInfo: ExecutionClusterInfo)
+    extends ResourceAllocator {
+
+  private val threshold = 40
+
+  private val historicalRuntimeStats = new HistoricalRuntimeStats()
+
+  private val operatorConfigs = new mutable.HashMap[PhysicalOpIdentity, OperatorConfig]()
+
+  private val linkPartitionInfos = new mutable.HashMap[PhysicalLink, PartitionInfo]()
+
+  private val linkConfigs = new mutable.HashMap[PhysicalLink, LinkConfig]()
+
+  private def avgHistoryPerOperator(physicalOpIdentity: PhysicalOpIdentity): (Double, Double) = {
+    val executionHistory = {
+      historicalRuntimeStats.getHistoricalRuntimeStatsByOperatorVersion(physicalOpIdentity)
+    }
+    if (executionHistory.isEmpty || executionHistory.length <= threshold)
+      return (0, 0)
+
+    val recordsByExecutionId: Map[Int, List[WorkflowRuntimeStatistics]] =
+      executionHistory
+        .filter(stats => Option(stats.getStatus).contains(1.toByte))
+        .groupBy(_.getExecutionId.intValue())
+        .view
+        .mapValues(_.toList)
+        .toMap
+
+    def trimLeadingZeroPairs(
+        records: List[WorkflowRuntimeStatistics]
+    ): List[WorkflowRuntimeStatistics] =
+      records.dropWhile(record =>
+        record.getInputTupleCnt.intValue() == 0 && record.getOutputTupleCnt.intValue() == 0
+      )
+
+    val filteredResult: List[WorkflowRuntimeStatistics] =
+      recordsByExecutionId.values.toList.flatMap(trimLeadingZeroPairs)
+
+    val workingTime = if (filteredResult.nonEmpty) {
+      filteredResult
+        .map(record =>
+          (2000000000 - record.getIdleTime.doubleValue() - record.getControlProcessingTime
+            .doubleValue()) / 1000000000 / record.getNumWorkers.intValue()
```

Comment:
what are the `2000000000` and `1000000000`? I don't get why are we having those constants. Also, if you need a constant, define it with a constant variable.